system/View/ViewDecoratorTrait.php

```diff
@@ -24,7 +24,7 @@ protected function decorateOutput(string $html): string
         $decorators = \config('View')->decorators;
 
         foreach ($decorators as $decorator) {
-            if (! is_subclass_of($decorator, ViewDecoratorInterface::class)) {
+            if (! in_array(ViewDecoratorInterface::class, class_implements($decorator), true)) {
```

Comment:
I think what we actually want here is the very underrated `is_a()`:
```suggestion
            if (! is_a($decorator, ViewDecoratorInterface::class, true)) {
```

Reply:
In this context, is_a() and is_subclass_of() do the same.
class_implements() only returns interfaces, but an additional array lookup is added.

I don't really see the point of these changes.

Reply:
Honestly, I forget about `is_a`, but gave it a whirl - and it doesn't work either. If you feel like giving my "MadLibs" example I put together for a new feature video last night a whirl, you try it yourself: https://github.com/lonnieezell/ci-tutorials/tree/4.2-decorators

Reply:
@lonnieezell In this repository, all three functions work the same way. No exceptions are thrown. windows, php 7.4.27, Spark 4.2

Reply:
I also have no problems: https://madlibs.gatner.be/madlib

Reply:
Super strange - is_a and is_subclass_of both throw the Invalid Decorator exception for me on an M1 Mac. 

Sounds like the safest thing is to go ahead with the change, though, so that we know it operates correctly everywhere.

Reply:
I also have no problems with PHP 8.0 and 7.4 on Intel Mac.

Reply:
No idea what was happening last night. Trashed the vendor folder on that project and reinstalled and it worked. Reverted this change.

Reply:
🤷‍♂️ When computers be like this, I always just say: keeps me employed!